src/build_system.ml

```diff
@@ -237,19 +237,24 @@ module Alias0 = struct
 end
 
 module Loaded = struct
-  type t = {
-    rules_produced : Rules.t;
-    targets : Path.Set.t;
-  }
+  type t =
+    | Non_build of Path.Set.t
+    | Build of {
+        rules_produced : Rules.t;
+        targets_here : Internal_rule.t Path.Map.t;
+        (* dir -> file -> rule *)
+        targets_of_forwarded_dirs : Internal_rule.t Path.Map.t Path.Map.t;
```

Comment:
I wonder if we couldn't simplify this a bit. The only use of `Forward` is for aliases. We could have two entries for the `Build` case: one for `_build/<dir>` and one for `_build/<dir>/.aliases`. Then when we fetch the rules of a directory, we would always lookup the directory without the `.aliases` and then choose which one of the two we want based on whether we are looking for an alias or not.

Reply:
Sure, that works. I'll make that change.